src/test/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControlTest.java

```diff
@@ -994,138 +1008,594 @@ void testGenerateAccessMetadataWithNoCaseId() {
+                                                                        boolean canAccessCaseType,
+                                                                        boolean canAccessCaseState) {
+
+            List<RoleAssignment> roleAssignments = roleNames.stream()
+                    .map(value -> RoleAssignment.builder()
+                        .roleName(ROLE_NAME_1)
+                        .grantType(STANDARD.name())
+                        .attributes(RoleAssignmentAttributes.builder().region(Optional.of("123")).build())
+                        .build()
+                    ).collect(Collectors.toList());
+
+            doReturn(roleAssignments)
+                .when(filteredRoleAssignments).getFilteredRoleAssignmentsFailedOnRegionOrBaseLocationMatcher();
+
+            setupAccessProfileCheckMocks(roleAssignments, outputAccessProfiles, canAccessCaseType, canAccessCaseState);
+
+            return roleAssignments;
+        }
+
+        private void addPseudoRoleAssignments(List<String> roleNames) {
+            roleNames.forEach(roleName ->
+                pseudoRoleAssignments.add(RoleAssignment.builder()
+                    .roleName(AccessControl.IDAM_PREFIX + roleName)
+                    .grantType(STANDARD.name())
+                    .build()
+                )
+            );
+        }
+
+        private List<RoleAssignment> filterRoleAssignmentsForAccessProfileServiceCall(
+            List<RoleAssignment> roleAssignments
+        ) {
+            return  roleAssignments.stream()
+                .filter(roleAssignment ->
+                    List.of(
+                        STANDARD.name(),
+                        SPECIFIC.name(),
+                        CHALLENGED.name()
+                    ).contains(roleAssignment.getGrantType())
+                ).collect(Collectors.toList());
+        }
 
-        doReturn(filteredRoleAssignments)
-                .when(roleAssignmentsFilteringService).filter(
-                    any(RoleAssignments.class), any(CaseDetails.class));
-        return defaultCaseDataAccessControl.generateAccessMetadata("CASE_ID");
     }
 
+
     private Map<String, JsonNode> generateCaseData(final String value)
         throws JsonProcessingException {
         final String caseData = (value == null) ? "{}"
             : "{\n" + "\"CaseAccessCategory\": \"" + value + "\"\n" + "  }";
         return JacksonUtils.convertValue(mapper.readTree(caseData));
     }
+
+
+    private static final String ACCESS_PROFILE_CHECK_NEGATIVE_MATCH_PARAMS
+        = "uk.gov.hmcts.ccd.domain.service.casedataaccesscontrol.DefaultCaseDataAccessControlTest"
+        + "#accessProfileCheckNegativeMatchParams";
+
+    @SuppressWarnings("unused")
+    private static Stream<Arguments> accessProfileCheckNegativeMatchParams() {
+        return Stream.of(
+            Arguments.of(List.of(), true, true),
+            Arguments.of(List.of("AP1"), false, true),
+            Arguments.of(List.of("AP2"), true, false)
```

Comment:
Add comment naming params here as to far away from usage to work it out looking at code.

Reply:
Resolved in 0b05cb7ec5928a0f7d297ed2a6ea340e9277af8f